### PR TITLE
ObjImporter can import obj files from the local disk as well as from the web.

### PR DESCRIPTION
**Pull request by LukasLiss**
The public import function does now first of all check whether the file is present on the current local file system. If so it will use the local import function. If not it will use the web URL based one.

### CODE REVIEW

**Review by BenediktHensen — CHANGES_REQUESTED** [review]
This duplicates a lot of code which makes maintenance more difficult.
I would suggest to stay with one method and to move the decision whether a Web request is necessary or a local file is loaded into a dedicated hybrid content loader. So, just add a class that implements the IContentLoader interface and handle the decision whether to load from file or the Web there. The IContentLoader does not even need a URL object so it should be able to work with file path strings.

This would also create a cleaner separation of concerns: The ObjImporter class is only responsible for administrating the import process but it should not care about the source where the object came from. It should just pass an address on to the content loader which is then able to check whether it is a local file or a Web file and provides the content from there.
The only modification that this class should probably get is to postpone the conversion of the string path into an URL to be as late as possible and only in the case that a Web request is actually made. The same is true for the MtlLibrary object. This one also needs to work with string paths and it should use the same hybrid content loader.

**Comment by LukasLiss** [comment]
As proposed I removed the duplicated code. Now there is only one function that handles all the imports for local and web-based files. As @BenediktHensen proposed, this function uses a string as input instead of an uri. This allowed handling local and web-based files in the same way. At first, I started working on the Contentloader class that handles local files differently than web-based ones, but after some research, I found out that the Unity Web request is already capable of doing so. Therefore the pushed code still uses the Unity Web request as a Content loader. 
The import of local and web-based files has been tested manually.

File: Assets/i5 Toolkit for Unity/Runtime/Importers/Scripts/ObjImporter.cs [hunk 1]
```diff
@@ -94,9 +93,9 @@ public async Task<GameObject> ImportAsync(string url)
                 // check that the referenced mtl library is already loaded; if not: load it
                 if (!MtlLibrary.LibraryLoaded(parseResult.LibraryPath))
                 {
-                    string mtlUri = UriUtils.RewriteFileUriPath(uri, parseResult.LibraryPath);
-                    string libraryName = System.IO.Path.GetFileNameWithoutExtension(uri.LocalPath);
-                    bool successful = await MtlLibrary.LoadLibraryAsyc(new Uri(mtlUri), libraryName);
+                    string mtlUri = System.IO.Path.ChangeExtension(path, ".mtl");
```
Comment (BenediktHensen) [inline Assets/i5 Toolkit for Unity/Runtime/Importers/Scripts/ObjImporter.cs:96]: This line changes the functionality and can lead to errors. The new version implies that the material file will always have the same name as the obj file and will be in the same directory. However, there are online repositories where the file is named differently or called differently. The original RewriteFileUriPath method takes the absolute path to the obj file and concatenates the relative library path.

The new version e.g. leads to errors when loading 3D model from https://people.sc.fsu.edu/~jburkardt/data/obj/obj.html.

Example: The file https://people.sc.fsu.edu/~jburkardt/data/obj/airboat.obj has its mtl library at ./vp.mtl. When trying to import it, it raises the follow error message:

grafik

Maybe something similar to this works for both URLs and local file paths?
`string absolutePath = Path.GetFullPath(baseDirectory + relativePath);`

**Review by BenediktHensen — CHANGES_REQUESTED** [review]
The changes are really good and resolve the previously made comment. 👍 

There is only one small thing as the pull request changes the way how the mtl path is constructed and the new way does not work with all files. Can you take a look at that? Maybe it is possible to concatenate the absolute path of the obj file with the relative path of the mtl path both for urls and local file paths?

**Review by BenediktHensen — APPROVED** [review]

**Pull request merged**